server/pyproject.toml

```diff
@@ -1,9 +1,34 @@
+[tool.poetry]
+name = "testflinger"
+description = "Testflinger Server"
+readme = "README.rst"
+version = "1.1.0"                  # specify your version here
+authors = []
+
+[tool.poetry.dependencies]
+python = "^3.10"                                        # specify your Python version requirement here
+apiflask = "^2.1.1"
+flask = "^3.0.3"
+flask-pymongo = "^2.3.0"
+gunicorn = "^22.0.0"
+gevent = "^24.2.1"
+prometheus-client = "^0.20.0"
+pyyaml = "^6.0.1"
+sentry-sdk = { extras = ["flask"], version = "^2.0.1" }
+
+[tool.poetry.dev-dependencies]
+pytest = "^8.1.2"
+
+[tool.poetry.scripts]
+testflinger-agent = "testflinger_agent.cmd:main"
+
+[build-system]
+requires = ["poetry-core>=1.0.0"]
+build-backend = "poetry.core.masonry.api"
+
 [tool.black]
 line-length = 79
 exclude = "charm/lib"
 
 [tool.pytest.ini_options]
-pythonpath = [
-  "charm/src",
-  "charm/lib"
-]
+pythonpath = ["charm/src", "charm/lib"]
```

Comment:
Interesting, does server tox here test even the charm code?

Reply:
True, I thought I had it set up to run from one of the github actions at one point but I don't see it now. I'll try to sort out somewhere to run this if things change with the charm in another PR. For now, the charm rarely changes though so I don't think it needs to rerun the unit tests in the charm for changes to the server code normally.